include/measurement_kit/dns/query.hpp

```diff
@@ -34,8 +37,8 @@ class Query {
     Query() {}
 
     /// Start an async DNS request.
-    Query(std::string query, std::string address,
-          std::function<void(Response)> func,
+    Query(QueryClass dns_class, QueryType dns_type, std::string name,
```

Comment:
I am not 100% convinced of the plan to use a struct to define the query type and class, but I will think a bit more about this.
Generally this is something I have found a bit annoying with other libraries since it means I need to know where to find the query types and names.
Perhaps we can overcome this by making an alternative constructor that takes the query name as a string.